<a href="https://colab.research.google.com/github/dorothyomare/ELDOHUB-CHALLENGE/blob/main/DOROTHY_OMARE_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from pandas.io.json import json_normalize
import requests
pd.set_option('max_rows', 99999)
pd.set_option('max_colwidth', 400)
pd.describe_option('max_colwidth')

display.max_colwidth : int or None
    The maximum width in characters of a column in the repr of
    a pandas data structure. When the column overflows, a "..."
    placeholder is embedded in the output. A 'None' value means unlimited.
    [default: 50] [currently: 400]


In [ ]:
URL = 'https://raw.githubusercontent.com/onaio/ona-tech/master/data/water_points.json'
#reading the url to data frame
df = pd.read_json(URL)
print(df)


    water_pay                 respondent  ...     simserial  subscriberid
0          no                  community  ...           NaN           NaN
1          no                  community  ...           NaN           NaN
2          no                  community  ...           NaN           NaN
3          no                  community  ...           NaN           NaN
4          no                  community  ...           NaN           NaN
5          no                  community  ...           NaN           NaN
6          no                  community  ...           NaN           NaN
7          no                  community  ...           NaN           NaN
8       na_dn                  community  ...           NaN           NaN
9          no                  community  ...           NaN           NaN
10         no                  community  ...           NaN           NaN
11         no                  community  ...           NaN           NaN
12         no                  communi

In [ ]:
data = pd.DataFrame(df)
data.head()

water_pay respondent  ... simserial subscriberid
0        no  community  ...       NaN          NaN
1        no  community  ...       NaN          NaN
2        no  community  ...       NaN          NaN
3        no  community  ...       NaN          NaN
4        no  community  ...       NaN          NaN

[5 rows x 48 columns]

In [ ]:
#previewing the botton rows of the table
data.tail(5)

water_pay                 respondent  ...     simserial  subscriberid
707        no                  community  ...           NaN           NaN
708        no  local_authority community  ...           NaN           NaN
709        no  local_authority community  ...           NaN           NaN
710        no                  community  ...  8.923301e+18  6.200101e+14
711        no                  community  ...           NaN           NaN

[5 rows x 48 columns]

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 712 entries, 0 to 711
Data columns (total 48 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   water_pay                    712 non-null    object        
 1   respondent                   712 non-null    object        
 2   research_asst_name           712 non-null    object        
 3   water_used_season            712 non-null    object        
 4   _bamboo_dataset_id           712 non-null    object        
 5   _deleted_at                  0 non-null      datetime64[ns]
 6   water_point_condition        712 non-null    object        
 7   _xform_id_string             712 non-null    object        
 8   other_point_1km              712 non-null    object        
 9   _attachments                 712 non-null    object        
 10  communities_villages         712 non-null    object        
 11  end                          712 non-null    

In [ ]:
essential_attr = data[['_id','communities_villages', 'water_point_condition']]
essential_attr.head(10)

_id communities_villages water_point_condition
0  381705            Gumaryili           functioning
1  381706            Selinvoya           functioning
2  381707            Selinvoya           functioning
3  381708            Selinvoya           functioning
4  381709            Selinvoya           functioning
5  381710            Selinvoya     newly_constructed
6  381711            Selinvoya           functioning
7  381712            Selinvoya     newly_constructed
8  381713            Selinvoya    under_construction
9  381714            Selinvoya           functioning

Number of functioning waterpoints

# Number of functioning waterpoints

In [ ]:
essential_attr.isnull().sum()

_id                      0
communities_villages     0
water_point_condition    0
dtype: int64

In [ ]:
def WaterFunctional(essential_attr):
    okay= essential_attr.water_point_condition.value_counts()
    return okay
WaterFunctional(essential_attr)

functioning           590
broken                 40
abandoned              36
newly_constructed      28
under_construction     16
na_dn                   2
Name: water_point_condition, dtype: int64

The number of functioning water points is 590

**The number of water points per community**,

# The number of water points per community,

In [ ]:
def WaterPerCommunity(essential_attr):
    summary =essential_attr.groupby(essential_attr["communities_villages"])['water_point_condition'].value_counts()
    return summary
    
WaterPerCommunity(essential_attr)

communities_villages  water_point_condition
Abanyeri              functioning               3
                      newly_constructed         1
Akpari-yeri           functioning               3
Alavanyo              functioning               2
                      abandoned                 1
Arigu                 functioning               6
                      abandoned                 4
                      broken                    2
Badomsa               functioning              26
                      newly_constructed         1
Bandem                functioning               4
                      broken                    2
                      newly_constructed         1
Banyangsa             functioning               8
                      broken                    1
                      na_dn                     1
Bechinsa              functioning              26
Chanpolinsa           functioning               4
Chansa                functioning               9
Chonde

#The rank for each community by the percentage of broken water points.

In [ ]:
def PercentageBrocken(essential_attr):
    counts =essential_attr.groupby(essential_attr["communities_villages"])['water_point_condition'].value_counts('broken')
    return counts
PercentageBrocken(essential_attr) 


communities_villages  water_point_condition
Abanyeri              functioning              0.750000
                      newly_constructed        0.250000
Akpari-yeri           functioning              1.000000
Alavanyo              functioning              0.666667
                      abandoned                0.333333
Arigu                 functioning              0.500000
                      abandoned                0.333333
                      broken                   0.166667
Badomsa               functioning              0.962963
                      newly_constructed        0.037037
Bandem                functioning              0.571429
                      broken                   0.285714
                      newly_constructed        0.142857
Banyangsa             functioning              0.800000
                      broken                   0.100000
                      na_dn                    0.100000
Bechinsa              functioning              1.000000
Chan

# Conclusion

In [ ]:
From the results, it is clear that most water points are functional. There is also a good number of broken water points are abandoned,broken,newly constructed or under construction.